proteus/SpatialTools.py

```diff
@@ -945,23 +949,47 @@ def __init__(self, domain, radius, coords, barycenter, nPoints):
         self.regions = np.array([[xc, yc]])
 
         # boundary tags
-        self.boundaryTags = bt = {'circle':1}
-        vertFlag=[]
-        segmFlag=[]
-        for kkk in range(nVertices):
-                vertFlag.append(bt['circle'],)
-                segmFlag.append(bt['circle'],)
-        self.vertexFlags = np.array(vertFlag)
-        self.segmentFlags = np.array(segmFlag)
-        self.facetFlags=np.array([1])
-        self.regionFlags = np.array([1])
-
-        # boundary list
-        self.BC = {'circle': self.BC_class(shape=self, name='circle',
-                                       b_or=self.b_or, b_i=0),
-                  }
-        self.BC_list = [self.BC['circle'],
-                       ]
+        if boundaryTag is None:
+            self.boundaryTags = bt = {'circle':1}
+            vertFlag=[]
+            segmFlag=[]
+            for kkk in range(nVertices):
```

Comment:
I think we can do better with this choice of index name. I don't have full context, but I'm quite sure anything else is better. For some of these loops, you can get away with using an underscore:

```
for _ in range(foo):
```

Reply:
Yes of course. I wasn't aware of the context (I didn't write this function), so I didn't change the variable name.